compass/landice/tests/mismipplus/setup_mesh.py

```diff
@@ -0,0 +1,479 @@
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+from mpas_tools.mesh.conversion import convert, cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+
+from compass.model import make_graph_file
+from compass.step import Step
+
+
+class SetupMesh(Step):
+    """
+    A step for creating a mesh and initial condition for MISMIP+ test cases
+    Experimental protocol described in:
+    Asay-Davis, et al. 2016. "Experimental Design for Three Interrelated
+    Marine Ice Sheet and Ocean Model Intercomparison Projects:
+    MISMIP v.3 (MISMIP+), ISOMIP v.2 (ISOMIP+) and MISOMIP v.1 (MISOMIP1)."
+    Geoscientific Model Development 9 (7): 2471–97.
+    https://doi.org/10.5194/gmd-9-2471-2016.
+
+    Attributes
+    ----------
+    resolution : int
+        The resolution of the test case, as defined in the configuration file
+        at the time when `compass setup` is run.
+
+    nx : int
+        The number of cells in the x direction
+
+    ny : int
+        The number of cells in the y direction
+
+    dc : int
+        The distance in meters between adjacent cell centers.
+
+    """
+    def __init__(self, test_case, name, subdir, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        name : str
+            the name of the test case
+
+        subdir : str
+            the subdirectory for the step.  The default is ``name``
+
+        resolution : int
+            The nominal distance [m] between horizontal grid points
+        """
+
+        super().__init__(test_case=test_case,
+                         name=name,
+                         subdir=subdir)
+
+        # Files to be created as part of the this step
+        for filename in ['mpas_grid.nc', 'graph.info', 'landice_grid.nc']:
+            self.add_output_file(filename=filename)
+
+        self.resolution = resolution
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+
+        # hard coded parameters
+        Lx = 640e3  # [m]
+        Ly = 80e3   # [m]
+
+        logger = self.logger
+        config = self.config
+
+        section = config['mesh']
+
+        # read the resolution from the .cfg file at runtime
+        resolution = section.getfloat('resolution')
+        # read the gutter lenth from the .cfg file
+        gutterLength = section.getfloat('gutter_length')
+        # ensure that the requested `gutterLength` is valid. Otherwise set
+        # the value to zero, such that the default `gutterLength` of two
+        # gridcells is used.
+        if (gutterLength < 2. * resolution) and (gutterLength != 0.):
+            gutterLength = 0.
+
+        # check if the resolution has been changed since the `compass setup`
+        # command was run
+        if self.resolution != resolution:
+            raise Exception(f'Resolution was set at {self.resolution:2d}m'
```

Comment:
I got this exception to be raised with the `-f` config testing I mentioned above.  In so doing, I got this value error for this line:
```
  File "/usr/projects/climate/mhoffman/compass/mismip+_spinup/compass/landice/tests/mismipplus/setup_mesh.py", line 94, in run
    raise Exception(f'Resolution was set at {self.resolution:2d}m'
ValueError: Unknown format code 'd' for object of type 'float'
```

Reply:
Thanks for testing this! I wasn't really aware of the `-f` option. 

I've revamped the way the directory structure is created (see response to your original comment). With that revamp, this bug should be fixed and configuration options passed from the command line should be properly parsed.  (Also the error you got in the `SmokeTest` should be fixed). 

All that was pushed in [4b0f35a](https://github.com/MPAS-Dev/compass/pull/737/commits/4b0f35a01e4606b80d1b86e37910645cfd780e92).